File: Element.scala
case class Element(var value: AnyRef)
{
  var leftChild: Element = null
  var rightChild: Element = null
  var parent: Element = null
  var weight: Int = 0

  def printElement() = println("Значение выбранного узла: " + value);

  def getParent() = this.parent;

  def setParent(element: Element) = this.parent = element;

  def getValue() = this.value;

  def setValue(value: AnyRef): Unit = this.value = value;

  def getWeight() =
  {
    try this.weight
    catch
    {
      case e: NullPointerException =>
        0
    }
  };

  def setWeight(weight: Int) = this.weight = weight;

  def getLeftChild() = this.leftChild;

  def setLeftChild(left: Element) = this.leftChild=left;

  def getRightChild() = this.rightChild;

  def setRightChild(right: Element) = this.rightChild = right;

  override def toString: String = "Узел{" + "Значение=" + value + '}'
}

File: Kotlin/Element.kt
class Element {

    var value: Any? = null

    var weight = 0

    var leftChild: Element? = null
    internal var rightChild: Element? = null
    var parent // Родитель узла
            : Element? = null

    fun Element() {
        leftChild = null
        rightChild = null
        parent = null
    }

    @JvmName("setWeight1")
    fun setWeight(weight: Int) {
        this.weight = weight
    }

    @JvmName("getLeftChild1")
    fun getLeftChild(): Element? {
        return leftChild
    }

    @JvmName("setLeftChild1")
    fun setLeftChild(leftChild: Element?) {
        this.leftChild = leftChild
    }

    fun getRightChild(): Element? {
        return rightChild
    }

    fun setRightChild(rightChild: Element?) {
        this.rightChild = rightChild
    }

    override fun toString(): String {
        return "Узел{" +
                "Значение=" + value +
                '}'
    }
}
